File: springboot-app/src/main/java/com/lumos/lumosspring/execution/dto/response/PreMeasurementResponseDTO.java
package com.lumos.lumosspring.execution.dto.response;

import java.util.List;

public record PreMeasurementResponseDTO(
        long preMeasurementId,
        long contractId,
        String city,
        String createdBy,
        String createdAt,
        String depositName,
        String preMeasurementType,
        String preMeasurementStyle,
        String teamName,
        String totalPrice,
        List<PreMeasurementStreetResponseDTO> streets
) {
}

File: angular-app/src/app/models/pre-measurement-response-d-t.o.ts
export interface PreMeasurementResponseDTO {
  preMeasurementId: number;
  contractId: number;
  city: string;
  createdBy: string;
  createdAt: string;
  depositName: String;
  preMeasurementType: string;
  preMeasurementStyle: string;
  teamName: string;
  totalPrice: string;
  status: string;

  streets: {
    number: number;
    preMeasurementStreetId: number;
    lastPower: string;
    latitude: number;
    longitude: number;
    street: string;
    hood: string;
    city: string;
    status: string;

    items: {
      preMeasurementStreetItemId: number;
      contractItemId: number;
      contractReferenceItemName: string;
      contractReferenceItemType: string;
      contractReferenceItemPower: string;
      contractReferenceItemLength: string;
      measuredQuantity: number;
      itemStatus: string;
    }[]

  }[];
}
